src/summary/profile/relationships-card.component.tsx

```diff
@@ -1,15 +1,55 @@
 import React from "react";
-import SummaryCard from "../cards/summary-card.component";
 import { match } from "react-router";
+import { openmrsFetch } from "@openmrs/esm-api";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import SummaryCard from "../cards/summary-card.component";
+import SummaryCardRow from "../cards/summary-card-row.component";
+import SummaryCardRowContent from "../cards/summary-card-row-content.component";
+import VerticalLabelValue from "../cards/vertical-label-value.component";
 
 export default function RelationshipsCard(props: RelationshipsCardProps) {
+  const [relationships, setRelationships] = React.useState(null);
+  React.useEffect(() => {
+    openmrsFetch(
```

Comment:
We've usually been putting these calls into a resource file for a couple reasons:

1. Easier mocking of the request

```js
jest.mock('./related-person.resource.tsx', () => ({
  getRelationships: jest.fn()
})
```

2. Reuse of any transformation logic that needs to be applied to the request/response objects when interacting with the RelatedPerson resource.

What are your thoughts on that? It looks like you've already solved (1) very easily just by mocking openmrsFetch itself. And for (2) we aren't really doing any transformation of the response object. So maybe okay not creating the resource files?

Reply:
hi @joeldenning , good points. I updated the PR and separated the call into a separate file to stick to the convention that we currently have.